File: chapter01/chapter01/node_tuto/scripts/minimal_subscriber.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float64

def callback(data):
	rospy.loginfo("I receive : %f",data,data)

def listener():
	rospy.init_node("minimal_subscriber")
	rospy.Subscriber("topic1", Float64, callback)
	rospy.spin()

File: chapter01/node_tuto/src/minimal_subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/Float64.h>

void myCallback(const std_msgs::Float64& msg){

	ROS_INFO("Receive = %f ", msg.data);
}

int main (int argc,char**argv){

	ros::init(argc,argv, "minimal_subscriber");
	ros::NodeHandle n;

	ros::Subscriber sub = n.subscribe("topic1", 1,myCallback);

	ros::spin();
	return 0;
}